api/models/log_event.py

```diff
@@ -94,14 +94,14 @@ class StudentLogEvent(LogEvent):
         Group,
         null=True,
         blank=True,
-        on_delete=models.CASCADE,
+        on_delete=models.SET_NULL,
```

Comment:
oh, right, I said `PROTECT` earlier, but that was wrong, of course. `SET_NULL` is what we need :)